modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/store/useLiferayStore.es.js

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+
+export default function useLiferayStore(name, reducer, initialState) {
+	let namedState = Liferay.State.get(name);
+
+	if (!namedState) {
+		namedState = Liferay.State.register(name, reducer, initialState);
+	}
+
+	const store = React.useMemo(() => namedState.store, [namedState]);
+
+	const [stateValue, setStateValue] = React.useState(store.getState());
+
+	React.useEffect(() => {
+		const unsub = store.subscribe(() => {
+			const newStore = store.getState();
+
+			setStateValue(newStore);
+		});
+
+		return unsub;
+	}, [store]);
+
+	return {
+		state: stateValue,
+		dispatch: store.dispatch,
+		replaceReducer: store.replaceReducer,
```

Comment:
I'm not actually sure this is needed, I think we can get away with just having the reducer as an argument and not even use this.